src/Confluent.Kafka/Error.cs

```diff
@@ -18,19 +18,30 @@
 
 namespace Confluent.Kafka
 {
-    public struct Error
+    public class Error
     {
         public Error(ErrorCode code)
         {
             Code = code;
+            reason = null;
+        }
+
+        public Error(ErrorCode code, string reason)
+        {
+            Code = code;
+            this.reason = reason;
         }
 
         public ErrorCode Code { get; }
 
-        // Note: In most practical scenarios there is no benefit to caching this +
-        //       significant cost in keeping the extra string reference around.
-        public string Message
-            => Internal.Util.Marshal.PtrToStringUTF8(Impl.LibRdKafka.err2str(Code));
+        private string reason;
+
+        // Rich error string, will be empty for APIs
+        // where there was just an ErrorCode. See ToString()
```

Comment:
Comment like `/// <summary></summary>` tags (see elsewhere for formatting).

Also, comment is wrong (it'll be rich error string or that returned by `ErrorCode.GetReason`